### PR TITLE
chore(testing-framework): a new behavior to print service call parameters/result out to console

### PR DESCRIPTION
**Pull request by raftedproc**
This patch introduces new test framework behavior to simplify AIR tests troubleshooting.

### CODE REVIEW

File: crates/testing-framework/src/asserts/behavior.rs [hunk 1]
```diff
@@ -66,7 +68,8 @@ impl Behavior {
         use Behavior::*;
 
         match self {
-            Echo => {
+            Echo => echo_call_service()(params),
+            DbgEcho => {
                 println!("echo_call_service() {:#?}", params);
```
[the remark this answers is not in the document]
Comment (monoid) [reply crates/testing-framework/src/asserts/behavior.rs:73]: You don't need to clone anything as `dbg!` just returns its argument.

[the remark this answers is not in the document]
Comment (monoid) [reply crates/testing-framework/src/asserts/behavior.rs:73]: You also may use `dbg!(&var);`.

File: crates/testing-framework/src/asserts/mod.rs [hunk 2]
```diff
@@ -54,6 +54,9 @@ pub enum ServiceDefinition {
     /// Some known service by name: "echo", "unit" (more to follow).
     #[strum_discriminants(strum(serialize = "behaviour"))]
     Behaviour(Behavior),
+    /// Same services as defined by the enum element above with dbg! applied to the arguments.
+    #[strum_discriminants(strum(serialize = "dbg_behaviour"))]
```
Comment (monoid) [inline crates/testing-framework/src/asserts/mod.rs:58]: I think calling it just `dbg` is more ergonomic.

File: air/tests/test_module/features/errors/error.rs [hunk 3]
```diff
@@ -29,9 +29,9 @@ fn fail_with_rebubble_error() {
                 (match 1 2 (null) )
                 (fail :error:)
             )
-            (call "peer_id" ("m" "f1") [:error:] scalar1) ; behaviour = echo
+            (call "peer_id" ("m" "f1") [:error:] scalar1) ; dbg_behaviour = echo
         )
-        (call "peer_id" ("m" "f2") [:error:] scalar2) ; behaviour = echo
+        (call "peer_id" ("m" "f2") [:error:] scalar2) ; dbg_behaviour = echo
```
Comment (monoid) [inline air/tests/test_module/features/errors/error.rs:34]: There are dedicated tests in the `crates/testing-framework/src/execution/mod.rs`.

File: crates/testing-framework/src/asserts/mod.rs [hunk 4]
```diff
@@ -103,6 +106,7 @@ impl ServiceDefinition {
                 call_map,
             } => call_seq_error(call_number_seq, call_map),
             ServiceDefinition::Behaviour(name) => name.call(params),
+            ServiceDefinition::DbgBehaviour(name) => name.call(dbg!(params)),
```
Comment (monoid) [inline crates/testing-framework/src/asserts/mod.rs:109]: Wouldn't be nice to dbg result too?

File: crates/testing-framework/src/asserts/mod.rs (hunk 2 above)
Comment (raftedproc) [reply crates/testing-framework/src/asserts/mod.rs:58]: But `dbg` tells nothing regarding the actual effect that happens.

Comment (mikevoronov) [reply crates/testing-framework/src/asserts/mod.rs:58]: I also vote for `dbg_behaviour`, it makes more sense for people who not familiar with Rust-specific `dbg` function.

File: crates/testing-framework/src/asserts/mod.rs (hunk 4 above)
Comment (mikevoronov) [reply crates/testing-framework/src/asserts/mod.rs:109]: Agreed

**Review by mikevoronov — APPROVED** [review]
utACK

**Pull request merged**